atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTimestampBoundStore.java

```diff
@@ -128,25 +136,38 @@ private void cas(Client client, Long oldVal, long newVal) {
                     ConsistencyLevel.SERIAL,
                     ConsistencyLevel.EACH_QUORUM);
         } catch (Exception e) {
+            log.error("[CAS] Error trying to set from {} to {}: {}", oldVal, newVal, e.getMessage());
             lastWriteException = e;
             throw Throwables.throwUncheckedException(e);
         }
         if (!result.isSuccess()) {
-            String msg = "Timestamp limit changed underneath us (limit in memory: " + currentLimit
+            String msg = "Unable to CAS from " + oldVal + " to " + newVal + ". "
+                    + "Timestamp limit changed underneath us (limit in memory: " + currentLimit
+                    + ", stored in DB: " + getCurrentTimestampValues(result)
                     + "). This may indicate that another timestamp service is running against this cassandra keyspace."
                     + " This is likely caused by multiple copies of a service running without a configured set of"
                     + " leaders or a CLI being run with an embedded timestamp service against an already running"
                     + " service.";
             MultipleRunningTimestampServiceError err = new MultipleRunningTimestampServiceError(msg);
             log.error(msg, err);
+            log.error("Thread dump: " + ThreadDumps.programmaticThreadDump());
             lastWriteException = err;
             throw err;
         } else {
+            log.trace("[CAS] Setting cached limit to {}.", newVal);
             lastWriteException = null;
```

Comment:
why does `lastWriteException` even exist?  It's only ever modified, not read.  If you remove the  `@GuardedBy("this")` in line 68/71 you'll see that the compiler will even warn you that it's never used.